crates/parser2/src/parser/expr.rs

```diff
@@ -164,9 +168,19 @@ fn infix_binding_power<S: TokenStream>(parser: &mut Parser<S>) -> Option<(u8, u8
         Amp => (100, 101),
         LShift | RShift => (110, 111),
         Plus | Minus => (120, 121),
-        Star | Slash | Percent => (130, 131),
+        Star | Slash | Percent => {
+            if is_aug(parser) {
```

Comment:
This is what I came up with, to give accurate BPs by peeking forward, is this approach concise/fast enough ? if so I'll do the same in the Plus | Minus and Lshift | Rshift arms.